### DISCUSSION TITLE
duplicate symbol: tds_invalid_funcs when linking ct-lib and db-lib.

### DISCUSSION THREAD

**Discussion by PoiXP** [FreeTDS/freetds#552] [Q&A] [answered]
DESCLAIMER: that we don't use make to build libraries and issue is likely related to our setup. Trying to understand the build configuration of the library.

My understanding that `duplicate symbol: tds_invalid_funcs` happens because both `dblib.c` and `ct.c` includes `data.h` that contains header implemented functions. So when both libraries linked the linker complains.

Could assist with how it works? Or db-lib and ct-lib should never be linked by one application?

**Comment by freddy77** [comment, marked as answer]
They are not designed to be built together. They provide the same database access, you should pick one. And code is designed to be packaged in shared object not all in a single executable/library. So if you compile correctly you could load them both without symbols clash.
